### PR TITLE
Add java runtime check when create/update deployment

### PR DESCRIPTION
**Pull request by zmssp**
1. Bypass jar check for enterprise tier
2. Check jar java version and asa java version
3. Add disable-app-log to disable app logs when create/deployment

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring

File: src/spring/azext_spring/_validators.py [hunk 1]
```diff
@@ -583,30 +583,43 @@ def _validate_route_table(namespace, vnet_obj):
...
+        raise InvalidArgumentValueError(
+            "Do not find MANIFEST.MF, please check if your artifact is a valid fat jar" + tips)
     if file_size / 1024 / 1024 < 10:
         telemetry.set_user_fault("invalid_jar_thin_jar")
         raise InvalidArgumentValueError("Thin jar detected, please check if your artifact is a valid fat jar" + tips)
-
+    version_number = int(runtime_version[len("Java_"):])
+    if jdk_version not in _java_runtime_in_number():
+        raise InvalidArgumentValueError("Your java application is compiled with {}, currently the supported "
+                                        "java version is Java_8. Java_11, Java_17, you can configure the java runtime "
```
Comment (yuwzho) [inline src/spring/azext_spring/_validators.py:616]:
```suggestion
                                        "java version is Java_8, Java_11, Java_17, you can configure the java runtime "
```

Comment (zmssp) [reply src/spring/azext_spring/_validators.py]: good catch, fixed

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
Peng will update the constraints in later PR

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**